packages/pigeon/example/app/android/app/src/main/java/io/flutter/plugins/Messages.java

```diff
@@ -356,9 +366,16 @@ public void error(Throwable error) {
   /** Generated class from Pigeon that represents Flutter messages that can be called from Java. */
   public static class MessageFlutterApi {
     private final @NonNull BinaryMessenger binaryMessenger;
+    private final String messageChannelSuffix;
 
     public MessageFlutterApi(@NonNull BinaryMessenger argBinaryMessenger) {
+      this(argBinaryMessenger, "");
+    }
+
+    public MessageFlutterApi(
+        @NonNull BinaryMessenger argBinaryMessenger, @NonNull String messageChannelSuffix) {
       this.binaryMessenger = argBinaryMessenger;
+      this.messageChannelSuffix = messageChannelSuffix;
```

Comment:
We can add it at this stage so we only have to have the logic in one spot.

Reply:
if only it were so simple